engines/bladerunner/subtitles.cpp

```diff
@@ -0,0 +1,778 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "bladerunner/subtitles.h"
+
+#if SUBTITLES_SUPPORT
+
+#include "bladerunner/bladerunner.h"
+#include "bladerunner/font.h"
+#include "bladerunner/text_resource.h"
+#include "bladerunner/audio_speech.h"
+//#include "bladerunner/script/scene_script.h" // for Game_Flag_Query declaration (actually script.h, but this seems to be included in other source files instead)
+#include "bladerunner/game_flags.h" // for Game_Flag_Query declaration (actually script.h, but this seems to be included in other source files instead)
+#include "bladerunner/game_constants.h" // for EDS flags - for subtitle checkbox flag state
+#include "common/debug.h"
+#include "common/util.h"
+
+namespace BladeRunner {
+
+/*
+ * Optional support for subtitles
+ * CHECK what happens in VQA where the audio plays separately (are the finales such VQAs ?)
+ *
+ * TODO in python script (FON from png glyphs) check if you can have semi-transparent pixels to better outline the fringe points of the glyphs - check what happens when MSB is set (transparency) and the rest of the color value is not all 0s.
+ * TODO Catch error for bad symbol in a quote (one that causes the font to crash) - this could happen with the corrupted internal font (TAHOMA18) -> font crash or bad font display / garbage character
+ * TODO add a keyboard shortcut key to enable / disable subtitles?
+ * TODO have a debug script to detect/report problematic lines (too long)
+ *
+ * TODO? put external FON and TRE in a new folder "SUBS" - case insensitive (?)
```

Comment:
Our filesystem code takes care of case sensitivity. Everything is lowercased before string comparison, so you can use any case.